pkg/utils/compatibility/batch.go

```diff
@@ -56,5 +53,8 @@ func discoverBatchAPICompatibility() {
 }
 
 func IsBatchV1CronJobSupported() bool {
+	once.Do(func() {
```

Comment:
Do we need to add `IsUnitTest` here?

once.Do(func() {
    if !testutil.IsUnitTest() {
        discoverBatchAPICompatibility()
    }
})